website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,490 @@
+
+While the default behavior will fit the needs of most organizations, there are occasions where this approach won't jive.
+
+For example, dbt expects that it has permission to create schemas on the fly (and we recommend that the users running dbt have this ability), but it might not be allowed at your company.
+
+Or, based on how you've designed your warehouse, you may wish to minimize the number of schemas in your dev environment (and avoid schema sprawl by not creating the combination of all developer schemas and custom schemas).
+
+Alternatively, you may even want your dev schemas to be named after feature branches instead of the developer name.
+
+For this reason, dbt offers three macros to customize what objects are created in the Data Warehouse:
+
+- [`generate_database_name()`](/docs/build/custom-databases#generate_database_name)
+- [`generate_schema_name()`](/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name)
+- [`generate_alias_name()`](/docs/build/custom-aliases#generate_alias_name)
+
+By overwriting one or multiple of those macros, we can tailor where dbt objects are created in the data warehouse and align with any existing requirement.
+
+:::note Key concept
+
+Models run from two different contexts must result in unique objects in the data warehouse. For example, a developer named Suzie is working on enhancements to `fct_player_stats`, but Darren is developing against the exact same object. In order to prevent overwriting each other's work, both Suzie and Darren should each have their unique versions of `fct_player_stats` in the development environment. Further, the staging version of `fct_player_stats` should exist in a unique location apart from the development versions, and the production version.
+
+:::
+
+
+We often leverage the following when customizing these macros:
+
+In dbt Cloud, we recommend utilizing [environment variables](/docs/build/environment-variables) to define where the dbt invocation is occurring (dev/stg/prod).
+
+They can be set at the environment level, and all jobs will automatically inherit the default values. We'll add jinja logic (`if/else/endif`) to identify whether the run happens in dev, prod, Ci and more so.
+    
+- Or as an alternative to environment variables, you can use `target.name`. For more information, you can refer to [About target variables](/reference/dbt-jinja-functions/target). 
+
+
+<Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/Environment Variables/custom-schema-env-var.png" title="Custom schema environmental variables target name." />
+
+To allow the database/schema/object name to depend on the current branch, you can use the out of the box `DBT_CLOUD_GIT_BRANCH` environment variable in dbt Cloud [special environment variables](/docs/build/environment-variables#special-environment-variables).
+
+
+## Example use cases
+
+Here are some typical examples we've encountered with dbt users leveraging those 3 macros and different logic. 
+
```

Comment:
i still think it'd be helpful to add bullets summarizing the diff use cases so users know what they are from the beginning of the page, rather than have them scroll. e.g:

- [Use custom schema without concatenating target schema in production](#use-custom-schema-without-concatenating-target-schema-in-production)
- etc...
etc...